src/adapter/src/catalog.rs

```diff
@@ -1480,6 +1485,10 @@ pub struct Source {
     pub desc: RelationDesc,
     pub timeline: Timeline,
     pub depends_on: Vec<GlobalId>,
+    pub custom_logical_compaction_window: Option<Duration>,
+    /// Whether the source's logical compaction window is controlled by
+    /// METRICS_RETENTION
+    pub is_retained_metrics_relation: bool,
```

Comment:
It seems a bit redundant to have both of these because the only two valid combinations are `(Some(_), true)` or `(None, false)`.

Reply:
That's true, but we might someday expose the ability to set logical compaction windows to users, but the corresponding objects would still have `is_retained_metrics_relation: false`